core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/Var.java

```diff
@@ -153,7 +153,9 @@ public boolean equals(Object o) {
 			return false;
 		}
 		Var var = (Var) o;
-		return anonymous == var.anonymous && Objects.equals(name, var.name) && Objects.equals(value, var.value);
+
+		return anonymous == var.anonymous && !(name == null && var.name != null || value == null && var.value != null)
+				&& Objects.equals(name, var.name) && Objects.equals(value, var.value);
```

Comment:
What makes this better? It reads to me as if it is just more complex. Doesnt `Objects.equals` take `null` into account?

Reply:
In the case that two Var objects are identical with the exception of one having a Value and the other not, then we can skip the equals check for the name.